### PR TITLE
feature(access_logs): make possible to specify an external s3 bucket …

### LINKED ISSUE #77: Use central logging bucket rather then create a new one.
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

We would like to have our ALB's log to a single bucket rather then individual ones.  Similar to the S3 pattern.  

## Expected Behavior

Pass a logging bucket in on creation.

## Use Case

Centralizing logging for a given env.

## Describe Ideal Solution

Extention of the module to support the current default but also accept an existing bucket like the s3 module does.

## Alternatives Considered

Writing our own as a fork from yours with this feature.

### PR DESCRIPTION
**Pull request by scream314**
…for logs

## what
* Added a new input variable to provide a bucket name for the logging bucket. If the name is provided, the `access_logs` module is skipped.

## why
* To be able to store multiple alb logs in a single bucket.
* To be able to provide an external bucket with a random name.

## references
* closes #77

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -98,6 +98,12 @@ variable "access_logs_enabled" {
   description = "A boolean flag to enable/disable access_logs"
 }
 
+variable "access_logs_s3_bucket_id" {
+  type        = string
+  default     = ""
```
Comment (SweetOps) [inline variables.tf:103]:
```suggestion
  default     = null
```

File: main.tf [hunk 2]
```diff
@@ -39,6 +39,8 @@ resource "aws_security_group_rule" "https_ingress" {
 }
 
 module "access_logs" {
+  count = var.access_logs_s3_bucket_id == "" ? 1 : 0
+
   source                             = "cloudposse/lb-s3-bucket/aws"
   version                            = "0.11.4"
   enabled                            = module.this.enabled && var.access_logs_enabled
```
Comment (SweetOps) [inline main.tf:46]: no reasons to add `count` you can just extend condition in the `enabled`

**Review by SweetOps — CHANGES_REQUESTED** [review]

File: main.tf [hunk 3]
```diff
@@ -39,6 +39,8 @@ resource "aws_security_group_rule" "https_ingress" {
 }
 
 module "access_logs" {
+  count = var.access_logs_s3_bucket_id == "" ? 1 : 0
```
Comment (adamantike) [inline main.tf:42]: GitHub doesn't allow me to comment on line 83 of this file, but that will have to be changed from:
```
bucket = module.access_logs.bucket_id
```
To:
```
bucket = coalesce(var.access_logs_s3_bucket_id, module.access_logs.bucket_id)
```

**Comment by adamantike** [comment]
@scream314, let us know if you can week working on this. I can take it, otherwise.

**Comment by scream314** [comment]
@adamantike I hope I can address the change requests next week.

**Comment by adamantike** [comment]
@SweetOps, I think this is ready for another review!

**Review by Gowiem — APPROVED** [review]

**Comment by Gowiem** [comment]
Thanks @scream314 -- released as `0.31.0`

**Pull request merged**